### PR TITLE
Remove post-review waffle and associated code

### LINKED ISSUE #6510: Remove post-review waffle, except in reviewer tools
post-review is live. We should remove usage of the waffle everywhere, except in reviewer tools for the time being. This will give us an off switch when `auto_approve` is having issues while not altering devhub pages etc.

### PR DESCRIPTION
**Pull request by diox**
Includes removal of:
- Old-style developer agreement
- Previous incarnation of auto-approve with conditions

Fix #6510

### CODE REVIEW

File: src/olympia/editors/models.py [hunk 1]
```diff
@@ -757,13 +757,6 @@ class AutoApprovalSummary(ModelBase):
     """Model holding the results of an auto-approval attempt on a Version."""
     version = models.OneToOneField(
         Version, on_delete=models.CASCADE, primary_key=True)
-    uses_custom_csp = models.BooleanField(default=False)
-    uses_native_messaging = models.BooleanField(default=False)
-    uses_content_script_for_all_urls = models.BooleanField(default=False)
-    average_daily_users = models.PositiveIntegerField(default=0)
-    approved_updates = models.PositiveIntegerField(default=0)
-    has_info_request = models.BooleanField(default=False)
-    is_under_admin_review = models.BooleanField(default=False)
```
Comment (EnTeQuAk) [inline src/olympia/editors/models.py]: Do we really want to get rid of them in respect to keeping historic data for a little while? Just asking, I don't have a strong opinion on that.

File: src/olympia/users/models.py [hunk 2]
```diff
@@ -156,6 +154,10 @@ class UserProfile(OnChangeMixin, ModelBase, AbstractBaseUser):
     # been compromised.
     auth_id = models.PositiveIntegerField(null=True, default=generate_auth_id)
 
+    # Date that the developer agreement last changed. Used to show the
+    # developer agreement to developers again when it changes.
+    last_developer_agreement_change = datetime(2017, 9, 22, 17, 36)
```
Comment (EnTeQuAk) [inline src/olympia/users/models.py]: I would actively describe in that comment that this is the date the "post-review" got enabled which simplifies interpreting that date at a later time when someone else might look at it.

File: src/olympia/editors/models.py (hunk 1 above)
Comment (diox) [reply src/olympia/editors/models.py]: Well, they are already unused now. They were only filled with the previous incarnation of auto-approval.

Comment (EnTeQuAk) [reply src/olympia/editors/models.py]: Then, nevermind and go on removing them :)

**Comment by EnTeQuAk** [comment]
If you rebase this branch to our current master, the CircleCI test should succeed.

**Pull request merged**